### PR TITLE
Allow "constructor" and other object properties as grammar literals

### LINKED ISSUE #520: Grammar with token "constructor" fails to generate
I noticed that the following grammar 

```
a = "constructor"
```

fails in version 4.0.2 with

```
error parsing grammar
Cannot read properties of undefined (reading 'length')
```

This was not the case in version 3.0.2

### PR DESCRIPTION
**Pull request by hildjj**
Fixes #520.

The `Intern` class uses an internal object to track whether a particular string has been seen before.  It checks this with `let num = this.offsets[s];`.  If one of the 8 base properties of `Object` is used as the literal name, `num` gets a non-numeric result.

The quick fix is to initialize `offsets` with an object that has no prototype instead of `{}`.

### CODE REVIEW

File: test/unit/compiler/passes/generate-bytecode.spec.js [hunk 1]
```diff
@@ -32,11 +32,27 @@ describe("compiler pass |generateBytecode|", () => {
...
         "b = 'b'",
         "c = 'c'",
+        "d = '__proto__'",
+        "e = 'constructor'",
+        "f = 'hasOwnProperty'",
+        "g = 'isPrototypeOf'",
+        "h = 'propertyIsEnumerable'",
+        "i = 'toLocaleString'",
+        "j = 'toString'",
+        "k = 'valueOf'",
```
Comment (Mingun) [inline test/unit/compiler/passes/generate-bytecode.spec.js:42]: Probably it is better to create a dedicated test with explicit intention to check that special properties are allowed.

**Comment by hildjj** [comment]
Puppeteer tests are broken again.  I'm going to take a few minutes to switch to playwright, and get the web tests running headless, cross-browser, and in CI.

File: test/unit/compiler/passes/generate-bytecode.spec.js (hunk 1 above)
Comment (hildjj) [reply test/unit/compiler/passes/generate-bytecode.spec.js]: I'll squash the two last commits if this change works for you @Mingun

**Review by Mingun — APPROVED** [review]

**Comment by Mingun** [comment]
I did not test changes by myself, but they seems correct, so rely on CI.

**Pull request merged**